briefcase/macos.py

```diff
@@ -52,6 +54,12 @@ def launcher_script_location(self):
         return macos_dir
 
     def build_app(self):
+        if self.dmg:
+            print("Building DMG file...")
+            dmg_name = self.formal_name.replace(' ', '_') + '.dmg'
+            dmg_path = os.path.join(os.path.abspath(self.dir), dmg_name)
+            dmgbuild.build_dmg(filename=dmg_path, volume_name=self.formal_name)
```

Comment:
This *creates* a dmg, but AFAICT, it doesn't put anything *into* the DMG. From a quick scan of the docs, we also need a `settings` kwarg, specifying at least a `files` key (pointing to the .app file). For bonus points, we could also include:
* `symlinks` (to provide the fast installation link to Applications)
* `icon` (to provide an icon for the volume)
* `background` (to provide a custom background image)
* `window_rect` (to set the initial size of the mounted window)

The DMG file should also be created in the macOS directory, so that all the macOS products end up in the same place.